vllm/v1/request.py

```diff
@@ -1,17 +1,28 @@
+from __future__ import annotations
+
 import enum
-from typing import TYPE_CHECKING, List, Optional, Union
+from functools import cached_property
+from typing import (TYPE_CHECKING, Any, Dict, List, Literal, Optional, Tuple,
+                    Union)
 
-from vllm.lora.request import LoRARequest
 from vllm.sampling_params import SamplingParams
 from vllm.sequence import RequestMetrics
 from vllm.v1.engine import EngineCoreRequest
 from vllm.v1.utils import ConstantList
 
 if TYPE_CHECKING:
+    from concurrent.futures import Future
+
+    from vllm.lora.request import LoRARequest
     from vllm.multimodal import MultiModalKwargs
     from vllm.multimodal.inputs import PlaceholderRange
+    from vllm.v1.core.guided_decoding import Grammar
     from vllm.v1.core.kv_cache_utils import BlockHashType
 
+GuidedDecodingObject = Union[str, Dict[str, Any]]
+GuidedDecodingKey = Tuple[Literal["json", "regex", "grammar", "choice"],
```

Comment:
should make `Literal["json", "regex", "grammar", "choice"]` from an enum so we can reference this list in other code